lib/prelude.dx

```diff
@@ -846,6 +834,73 @@ instance Ix Bool
     True -> 1
   unsafe_from_ordinal = \i. i > 0
 
+'## Subset class
+
+interface Subset a_sub a
+  inject' : a_sub -> a
+  project' : a -> Maybe a_sub
+
+def project {a} (a_sub:Type) [Subset a_sub a] (x:a) : Maybe a_sub =
+  project' x
+
+def inject {a_sub} (a:Type) [Subset a_sub a] (x:a_sub) : a =
```

Comment:
Why do we need this one? is it because no or all typeclass args have to be explicit?

Reply:
Yes, exactly